flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStoreFactory.java

```diff
@@ -0,0 +1,205 @@
+                            context.getCatalogTable().copy(filterLogStoreOptions(enrichedOptions)),
+                            context.getEnrichmentOptions(),
+                            context.getConfiguration(),
+                            context.getClassLoader(),
+                            context.isTemporary());
+            LogStoreTableFactory.discoverLogStoreFactory(
+                            Thread.currentThread().getContextClassLoader(),
+                            KafkaLogStoreFactory.IDENTIFIER)
+                    .onDropTable(logStoreContext, ignoreIfNotExists);
+        }
+    }
+
+    @Override
+    public Map<String, String> onCompactTable(
+            Context context, CatalogPartitionSpec catalogPartitionSpec) {
+        // TODO: next commit
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Set<ConfigOption<?>> requiredOptions() {
+        return Collections.emptySet();
+    }
+
+    @Override
+    public Set<ConfigOption<?>> optionalOptions() {
+        LogStoreTableFactory logStoreTableFactory =
+                LogStoreTableFactory.discoverLogStoreFactory(
+                        Thread.currentThread().getContextClassLoader(),
+                        KafkaLogStoreFactory.IDENTIFIER);
+        Set<ConfigOption<?>> options = new HashSet<>(logStoreTableFactory.optionalOptions());
+        options.addAll(logStoreTableFactory.requiredOptions());
+        options.add(BUCKET);
+        options.add(MANIFEST_TARGET_FILE_SIZE);
+        options.add(CHANGE_TRACKING);
+        return options;
+    }
+
+    // ~ Tools ------------------------------------------------------------------
+
+    private Map<String, String> filterLogStoreOptions(Map<String, String> enrichedOptions) {
+        Map<String, String> logStoreOptions = new HashMap<>();
+        LogStoreTableFactory logStoreTableFactory =
+                LogStoreTableFactory.discoverLogStoreFactory(
+                        Thread.currentThread().getContextClassLoader(),
+                        KafkaLogStoreFactory.IDENTIFIER);
+        Set<String> logStoreOptionKeys =
+                Stream.concat(
+                                logStoreTableFactory.requiredOptions().stream()
+                                        .map(ConfigOption::key),
+                                logStoreTableFactory.optionalOptions().stream()
+                                        .map(ConfigOption::key))
+                        .collect(Collectors.toSet());
+        enrichedOptions.forEach(
+                (k, v) -> {
+                    if (logStoreOptionKeys.contains(k)) {
+                        logStoreOptions.put(k, v);
+                    }
+                });
+        return logStoreOptions;
+    }
+
+    private static Path tablePath(Map<String, String> options, ObjectIdentifier identifier) {
+        return new Path(new Path(options.get(FILE_PATH.key())), identifier.asSummaryString());
```

Comment:
I am thinking we can create this path like: `root/catalogName.catalog/databaseName.db/tableName`.